src/web/client/services/graphClientService.ts

```diff
@@ -0,0 +1,152 @@
+        let requestUrl;
+
+        switch (service) {
+            case Constants.GraphService.GRAPH_MAIL:
+                requestUrl = new URL(userId, basePath);
+                break;
+            case Constants.GraphService.GRAPH_PROFILE_PICTURE:
+                requestUrl = new URL(
+                    path.join(
+                        userId,
+                        Constants.MICROSOFT_GRAPH_PROFILE_PICTURE_SERVICE_CALL
+                    ),
+                    basePath
+                );
+                break;
+            default:
+                return;
+        }
+
+        try {
+            WebExtensionContext.telemetry.sendAPITelemetry(
+                requestUrl.href,
+                service,
+                Constants.httpMethod.GET,
+                this.requestGraphClient.name
+            );
+
+            requestSentAtTime = new Date().getTime();
+
+            const response =
+                await WebExtensionContext.concurrencyHandler.handleRequest(
+                    requestUrl.href,
+                    {
+                        headers: {
+                            ...getCommonHeaders(this._graphToken),
+                        },
+                    }
+                );
+
+            if (!response.ok) {
+                throw new Error(JSON.stringify(response));
+            }
+
+            WebExtensionContext.telemetry.sendAPISuccessTelemetry(
+                requestUrl.href,
+                service,
+                Constants.httpMethod.POST,
+                new Date().getTime() - requestSentAtTime,
+                this.requestGraphClient.name
+            );
+
+            return await response.json();
+        } catch (error) {
+            const errorMsg = (error as Error)?.message;
+            if ((error as Response)?.status > 0) {
+                WebExtensionContext.telemetry.sendAPIFailureTelemetry(
+                    requestUrl.href,
+                    service,
+                    Constants.httpMethod.GET,
+                    new Date().getTime() - requestSentAtTime,
+                    this.requestGraphClient.name,
+                    errorMsg,
+                    "",
+                    (error as Response)?.status.toString()
+                );
+            } else {
+                WebExtensionContext.telemetry.sendErrorTelemetry(
+                    telemetryEventNames.WEB_EXTENSION_GET_FROM_GRAPH_CLIENT_FAILED,
+                    this.requestGraphClient.name,
+                    (error as Error)?.message,
+                    error as Error
+                );
+            }
+        }
+    }
+
+    public async getUserEmail(userId: string) {
+        try {
+            if (!this._graphToken) {
+                await this.graphClientAuthentication(true);
+            }
+
+            const response = await this.requestGraphClient(
+                Constants.GraphService.GRAPH_MAIL,
+                userId
+            );
+            return await response.mail;
```

Comment:
expect this to  be empty string in case of graph auth failures or API call faiures - we need to hook to the right fallback experience in that case